### ISSUE TITLE
Key board shurtcuts not working when mintmenu or other applets are open

### ISSUE THREAD

**Issue by Piiit** [linuxmint/cinnamon#5295]
To reproduce the issue:
1. Click on the sound applet or open the mint menue, or maybe also any other cinnamon applet (but this I haven't tested yet)
2. Use keyboard shortcuts, or multimedia keys, which will not work correctly 

What happens? Just some possibilities I tried...
- pressing left superkey used for the menu is triggered with some delay (around 2 sec.)
- pressing left superkey + D for desktop, opens the mint menu instead
- pressing multimedia key for sleep, does not trigger any operation at all
- pressing multimedia key for start/pause play, works only if the sound applet has been clicked and is open, when it is closed the music starts and after 1 sec. it stops again (It seems that the key click event is triggered twice with some delay)
- with sound applet open: mediakeys for volume work correctly (osd shown), but brightness (no osd) and sleep mode does not work 

Please note, that all this keyboard shortcuts and mediakeys work correctly without any applet open.

The overall impression is that the whole behaviour is not consistent, some key-down events do not work correctly as for the superkey.

Hope my explanation was clear and helps to solve this issue. 
Thanks for all your effort....

**Comment by NoXPhasma** [comment]
I can confirm this, happens to every applet. Also happens if you just open the context menu on the panel or any panel applet.

**Comment by collinss** [comment]
At least some of this has been an issue for quite a while (see https://github.com/linuxmint/Cinnamon/issues/3485). That being said, I've been poking around in the code to try to track it down again. The good news is I'm almost certain I know exactly what's causing the problem with the super key shortcuts and am working on a fix right now. The bad news is that the code that's causing the issues is complicated enough that this might have to wait for Cinnamon 3.2, as there is a strong possibility of regression elsewhere. (The code in question acts as a kind of switchboard for all events - keyboard and mouse - for any situation where Cinnamon intercepts these - ie menus, Cinnamon dialogs (such as the alt-f2 run dialog), scale and expo, and possibly some others.)

**Comment by i23098** [comment]
I'm on cinnamon 3.4.6 (on Fedora 26) and this bug is still occurring :( Can't lock screen using keys if some applet is open (e.g. menu, weather, music, etc)

**Comment by H8to** [comment]
No trying to necro bump here but, somebody seems to explain why this bug is there and even more interestingly the context menu of Firefox does not show the behaviour. 
The author of  the post points towards keyboard grabbing to be the issue and some toolkits have different ways of implementing it:
https://unix.stackexchange.com/questions/347337/why-keyboard-shortcuts-do-not-work-when-a-sub-menu-is-activated 

Maybe this hints into the right direction :smile:

**Comment by icarter09** [comment]
@collinss any update on this issue?

**Comment by collinss** [comment]
@icarter09 No, not really. The code in question needs a complete redesign, but I never got to it.

**Comment by camypaj** [comment]
I've seen this happen a few times, (if it's the same thing causing it), and it gets really annoying if a user clicks an applet, for example a calendar, and then uses the shortcut to lock the screen (ctrl+alt+L), nothing happens, and the screen stays unlocked.. :(

**Comment by sunnyjaiswal22** [comment]
This bug is present for a long time. I mostly encounter this when I try to lock the screen while the calendar applet is open. It would be great if this bug is fixed.

**Comment by ponjadito23** [comment]
This bug is still present on the latest version, don't close.
